tests/apollo/test_skvbc_chaotic_startup.py

```diff
@@ -0,0 +1,277 @@
+            expected=lambda v: v == initial_primary,
+            err_msg="Make sure we are in the initial view "
+        )
+
+        self.assertTrue(initial_view == view)
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(without=early_replicas),
+            checkpoint_num=1,
+            verify_checkpoint_persistency=False,
+            assert_state_transfer_not_started=False
+        )
+        checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+        # Verify the system is able to make progress
+        self.assertTrue(checkpoint_after > checkpoint_before)
+        
+        # Verify replicas that have initiated View Change catch up on state
+        await bft_network.wait_for_state_transfer_to_start()
+        for r in early_replicas:
+            await bft_network.wait_for_state_transfer_to_stop(initial_primary,
+                                                              r,
+                                                              stop_on_stable_seq_num=True)
+        print("STATUS: Early replicas that have initiated View Change catch up on state.")
+
+        # Stop one of the later started replicas, but not the initial Primary
+        # in order to verify that View Change will happen due to inability to
+        # reach quorums in the slow path. Also don't stop next primary, because
+        # in this test we check a single view change.
+        replica_to_stop = random.choice(bft_network.all_replicas(without=early_replicas | {initial_primary,
+                                                                                           expected_next_primary}))
+        print("STATUS: Stopping one of the later replicas with ID={}".format(replica_to_stop))
+        bft_network.stop_replica(replica_to_stop)
+
+        # Wait for View Change to happen.
+        view = await bft_network.wait_for_view(
+            replica_id=expected_next_primary,
+            expected=lambda v: v == expected_next_primary,
+            err_msg="Make sure we are in the next view "
+        )
+
+        self.assertTrue(expected_next_view == view)
+
+        await trio.sleep(10)  # TODO: remove when bft_network.wait_for_view also waits for system liveness.
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=expected_next_primary)
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(without={replica_to_stop}),
+            checkpoint_num=1,
+            verify_checkpoint_persistency=False,
+            assert_state_transfer_not_started=False
+        )
+        checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=expected_next_primary)
+
+        # Verify that the system is making progress after the View Change.
+        self.assertTrue(checkpoint_after > checkpoint_before)
```

Comment:
I would add one extra check that verifies that every single replica is in view 1, indicating that everyone is caught up together, and another single replica isn't lagging behind. We don't expect this ever to happen, but may as well check.